src/assets/scripts/resources-utils.js

```diff
@@ -0,0 +1,282 @@
+	// pagination templates written in nunjucks and vue: https://www.11ty.dev/docs/pagination/#paging-an-array
+	const pagination = {
+		items: dataInChunk[pageInQuery - 1],
+		pageNumber: pageInQuery - 1,
+		hrefs: hrefs,
+		href: {
+			next: hrefs[pageInQuery] ? hrefs[pageInQuery] : null,
+			previous: hrefs[pageInQuery - 2] ? hrefs[pageInQuery - 2] : null,
+			first: hrefs[0],
+			last: hrefs[hrefs.length - 1]
+		},
+		pages: dataInChunk,
+		page: {
+			next: dataInChunk[pageInQuery] ? dataInChunk[pageInQuery] : null,
+			previous: dataInChunk[pageInQuery - 2] ? dataInChunk[pageInQuery - 2] : null,
+			first: dataInChunk[0],
+			last: dataInChunk[dataInChunk.length - 1]
+		},
+		hideProceedingPageButton: pageInQuery !== 2 && pageInQuery !== 3,
+		hideFollowingPageButton: pageInQuery !== dataInChunk.length - 1 && pageInQuery !== dataInChunk.length - 2
+	};
+	return pagination;
+}
+
+/*
+ * Set the given list of checkbox values to 'checked' state.
+ *
+ * @param {Object} container - The container that has all checkbox elements
+ * @param {String} checkboxPrefix - The prefix of the checkbox name
+ * @param {Array} checkedValue - An array of the suffix of the checkbox name that are checked
+ */
+function renderCheckboxStats(container, checkboxPrefix, checkedValue) { // eslint-disable-line no-unused-vars
+	checkedValue.forEach(value => {
+		container.querySelector('input[type="checkbox"][name="' + checkboxPrefix + value + '"]').checked = true;
+	});
+}
+
+/*
+ * Set number of applied filters next to the filter header
+ *
+ *  @param {Object} container - The container within a filter header, where the number should be rendered
+ *  @param {String} checkboxPrefix - The prefix of the checkbox name
+ */
+function renderNumberOfAppliedFilters(container, checkboxPrefix) { // eslint-disable-line no-unused-vars
+	const appliedFilter = document.querySelectorAll('.filter-checkbox[name^="' + checkboxPrefix + '"]:checked');
+	container.innerHTML = appliedFilter.length > 0 ? '(' + appliedFilter.length + ')' : '';
+}
+
+/*
+ * Remove html tags from the input string.
+ * @param {String} inputString - The string to remove html tags.
+ * @return The string with html tags removed.
+ */
+// eslint-disable-next-line
+function stripHtmlTags(inputString) {
+	return inputString.replace(/<\/?[^>]+(>|$)/g, '');
+}
+
+/*
+ * Extract text from a html string.
+ * @param {String} input - The string to extract text from.
+ * @return The extracted text.
+ */
+function htmlDecode(input) {
+	let el = document.createElement('div');
+	el.innerHTML = input;
+	return el.innerText;
+}
+
+/*
+ * Process each object in the data set to convert some field value to formats for display.
+ * By default sort by publishedYear. 
+ * @param {Array<Object>} dataSet - The data set to process.
+ * @return The same set of the data set with fields converted.
+ */
+function processResourcesDisplayResults(inArray) { // eslint-disable-line no-unused-vars
```

Comment:
Rename `inArray` parameter name to `resources` as this function only makes sense with handling resources data.

Moreover, fix the code comment for this function. This comment applies to all other functions.